include/io/io_mesh_ascii.tcc

```diff
@@ -530,3 +530,47 @@ std::vector<std::tuple<mpm::Index, unsigned, double>>
   }
   return forces;
 }
+
+//! Return math function
+template <unsigned Tdim>
+std::vector<std::tuple<double, double>>
+    mpm::IOMeshAscii<Tdim>::read_math_function(
+        const std::string& math_function_file) {
+
+  // initialize math function
+  std::vector<std::tuple<double, double>> math_function;
+  math_function.clear();
+
+  // input file stream
+  std::fstream file;
+  file.open(math_function_file.c_str(), std::ios::in);
+
+  try {
+    if (file.is_open() && file.good()) {
+      // Line
+      std::string line;
+      while (std::getline(file, line)) {
+        boost::algorithm::trim(line);
+        std::istringstream istream(line);
+        // ignore comment lines (# or !) or blank lines
+        if ((line.find('#') == std::string::npos) &&
+            (line.find('!') == std::string::npos) && (line != "")) {
+          while (istream.good()) {
+            // x value
+            double x_value;
+            // fx value
+            double fx_value;
```

Comment:
You might want to put these two lines outside the while loop so that they are not generated and deleted multiple times.

Reply:
I prefer that with it being empty, because we are reading `istream`

Reply:
I don't think we need them to be empty. I modified the other `istream` read in all the read `io_mesh_ascii` functions in PR #680 as suggested by @kks32.

Reply:
Fair enough! Done and thanks!